### DISCUSSION TITLE
Do the points from the result of ecg_delineate match to original ECG?

### DISCUSSION THREAD

**Discussion by okovalchuk98** [neuropsychology/NeuroKit#762] [Q&A]
Hello,

My task is to detect amplitude and fragments of peaks in input ECG signal.

According to the [documentation](https://neuropsychology.github.io/NeuroKit/functions/ecg.html#ecg-delineate), the _ecg_delineate_ method takes as parameter cleaned ECG signal by the _ecg_clean_ method.
The _ecg_clean_ method may modify input ECG signal and the _ecg_delineate_ find features of ECG based on **cleaned** signal.

Do the points from the result of the _ecg_delineate_ method that uses **cleaned** signal match the points in the original (input) signal?
May I take the results of the _ecg_delineate_ method and apply it to the original input ECG signal and be sure that everything is correct?

Thanks

**Comment by DominiqueMakowski** [comment]
You can feed a raw signal to `ecg_delineate()` to run the algorithm on the raw signal, especially if you have a fairly clean signal. 
If you use a cleaned signal, you can still use the indices of the points of interest, as the x-axis is invariant between a cleaned and a raw signal (i.e., the 1000th point remains the 1000th point)
